build.gradle

```diff
@@ -12,33 +12,61 @@ repositories {
 
 test {
   systemProperty 'groovy.grape.enable', 'false'
-  useJUnitPlatform()
+
+  reports {
+    html.required = true
+    html.outputLocation = file("$project.buildDir/reports/html")
+  }
+  ignoreFailures = true
+
+  // show test results
+  def results = []
+  afterSuite { desc, result ->
+    if (desc.className) { results << result }
+  }
+
+  // show summary
+  doLast {
+    println "Tests: ${results.sum { it.testCount }}" +
+      ", Failures: ${results.sum { it.failedTestCount }}" +
+      ", Errors: ${results.sum { it.exceptions.size() }}" +
+      ", Skipped: ${results.sum { it.skippedTestCount }}"
+  }
 }
 
 compileGroovy {
   groovyOptions.forkOptions.jvmArgs = [ '-Dgroovy.grape.enable=false' ]
+  // @CNP annotation comes from the Jenkins Runtime instance
+  exclude '**/pluginActive.groovy'
 }
+
 compileTestGroovy {
   groovyOptions.forkOptions.jvmArgs = [ '-Dgroovy.grape.enable=false' ]
 }
 
 dependencies {
-  implementation group: 'org.codehaus.groovy', name: 'groovy-all', version: '2.5.22'
-  implementation group: 'org.codehaus.groovy', name: 'groovy-dateutil', version: '2.5.22'
+  implementation ('org.codehaus.groovy:groovy-all:2.4.21') {
+    exclude group: 'org.codehaus.groovy', module: 'groovy-ant'
+  }
 
   implementation group: 'com.cloudbees', name: 'groovy-cps', version: '3773.v505e0052522c'
   testImplementation group: 'org.apache.commons', name: 'commons-lang3', version: '3.12.0'
 
-  implementation group: 'org.jenkins-ci.main', name: 'jenkins-core', version: '2.409'
-  implementation group: 'org.jenkins-ci.plugins.workflow', name: 'workflow-step-api', version: '625.vd896b_f445a_f8', ext: 'jar'
-  implementation group: 'org.jenkinsci.plugins', name: 'pipeline-model-definition', version: '2.2169.vee7cd0efc13e', ext: 'jar'
+  implementation (group: 'org.jenkins-ci.main', name: 'jenkins-core', version: '2.460') {
+      exclude group: "org.codehaus.groovy", module: "groovy-all"
```

Comment:
hmm we can probably just pull the groovy dependency from Jenkins and remove the dependency on line 48?